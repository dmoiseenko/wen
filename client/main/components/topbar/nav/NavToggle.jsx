import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';


export default function NavToggle({ openCloseMenu, isOpen }) {
  function handleClick() {
    openCloseMenu();
  }

  const className = classnames('nav-toggle', {
    'is-active': isOpen
  });

  return (
    <span className={className} onClick={handleClick}>
      <span />
      <span />
      <span />
    </span>
  );
}

NavToggle.propTypes = {
  openCloseMenu: PropTypes.func.isRequired,
  isOpen: PropTypes.bool.isRequired
};
