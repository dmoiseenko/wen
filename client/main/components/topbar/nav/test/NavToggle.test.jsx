import React from 'react';
import { shallow } from 'enzyme';

import NavToggle from '../NavToggle';

const props = {
  openCloseMenu: jest.fn(),
  isOpen: false
};


it('should render closed if isOpen false', () => {
  const wrapper = shallow(
    <NavToggle
      {...props}
    />
  );

  expect(wrapper.node).toMatchSnapshot();
});

it('should render opened if isOpen true', () => {
  const wrapper = shallow(
    <NavToggle
      {...props}
      isOpen
    />
  );

  expect(wrapper.node).toMatchSnapshot();
});

it('should call openCloseMenu when click fires', () => {
  const openCloseMenu = jest.fn();

  const wrapper = shallow(
    <NavToggle
      {...props}
      openCloseMenu={openCloseMenu}
    />
  );

  wrapper.find('.nav-toggle').simulate('click');

  expect(openCloseMenu).toHaveBeenCalled();
});
