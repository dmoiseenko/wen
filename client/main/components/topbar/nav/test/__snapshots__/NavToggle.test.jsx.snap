// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render closed if isOpen false 1`] = `
<span
  className="nav-toggle"
  onClick={[Function]}
>
  <span />
  <span />
  <span />
</span>
`;

exports[`should render opened if isOpen true 1`] = `
<span
  className="nav-toggle is-active"
  onClick={[Function]}
>
  <span />
  <span />
  <span />
</span>
`;
